demisto_sdk/commands/content_graph/objects/pack.py

```diff
@@ -295,60 +317,32 @@ def set_content_items(self):
 
     def dump_metadata(self, path: Path, marketplace: MarketplaceVersions) -> None:
```

Comment:
there's a lot of logic here which is a bit hard to follow - please add a docstring with a detailed description, I will go over this afterward